artifactory/commands/transferfiles/utils.go

```diff
@@ -119,21 +121,29 @@ func pollUploads(phaseBase *phaseBase, srcUpService *srcUserPluginService, uploa
 		if progressbar != nil {
 			progressbar.SetRunningThreads(curProcessedUploadChunks)
 		}
-		curTokensBatch.fillTokensBatch(uploadTokensChan)
-
-		if len(curTokensBatch.UuidTokens) == 0 {
+		// Each uploading thread receive a token from the source via the uploadTokensChan, so this go routine can poll on it's status
+		fillTokensBatch(awaitingStatusChunksSet, uploadTokensChan)
+		// awaitingStatusChunksSet is used to keep all the uploaded chunks tokens in order to request their upload status from the source
```

Comment:
* Let's add a new-line before the beginning of this description.
* Missing period at the end of the sentence.

Reply:
done